cpp/src/io/parquet/writer_impl.cu

```diff
@@ -705,11 +1013,16 @@ void writer::impl::write_chunked(table_view const &table, pq_chunked_state &stat
       ck->start_row      = start_row;
       ck->num_rows       = (uint32_t)state.md.row_groups[global_r].num_rows;
       ck->first_fragment = i * num_fragments + f;
```

Comment:
I'm surprised this works: prior to lists, there was an assumption that the number of fragments was the same for every column (due to num_values==num_rows), but that no longer works for lists where the number of fragments can be different per column (The 2D grid num_fragments x num_column is no longer appropriate, and probably needs to be converted to a 1D fragment grid and store the column id or chunk id in the fragment struct).

Reply:
No, `num_fragments` is still the same for all columns. A fragment still contains 5000 rows. Then in initFragments, we translate list row index to value index in the leaf column. This means fragments can have different data sizes even for fixed width types now. I figured this would be ok because pages are still calculated based on fragment data sizes.

Yesterday, I realized that the number of values (not just rows) in a fragment also need to be capped to 5000 for the dictionary calculation to work, because of the fixed size of `s->dict`. This means that the method of fragment calculation needs to be updated (possibly split into two - one for row group size calculation and one for dictionary creation). My short term solution is to temporarily disable dictionary encoding for list types. I'm hoping to fix this in the refactor of parquet writer.